tools/globcheck/globcheck.py

```diff
@@ -0,0 +1,147 @@
+import os
+import sys
+import argparse
+import warnings
+import pandas as pd
+import logging as log
+import MDAnalysis as mda
+from Bio.PDB import PDBParser
+from cath_alphaflow.models.domains import GeneralDomainID, ChoppingPdbResLabel
+from cath_alphaflow.commands.measure_globularity import (
+    calculate_normed_radius_of_gyration,calculate_packing_density)
+
+# Basic logging configuration
+log.basicConfig(level=log.INFO, \
+                format='%(levelname)s - %(message)s')
+
+# Thresholds
+asphericity_threshold = 0.1
+rg_threshold = 0.356
+pd_threshold = 10.333
+
+def validate_residue_range(pdb_file, trimming_range):
+    """Validates that the provided residue trimming range is 
+    within the actual residue range present in the PDB structure.
+
+    Args:
+        pdb_file (str): Path to the input PDB
+        trimming_range (str): Residue range in format "start-end" ("1-122")
+    """
+    if not os.path.isfile(pdb_file):
+        log.error(f"❌ PDB file '{pdb_file}' does not exist.")
+        sys.exit(1)
+
+    start_res, end_res = map(int, trimming_range.split('-'))
+    u = mda.Universe(pdb_file)
+    residue_ids = [res.resid for res in u.residues]
+    min_res, max_res = min(residue_ids), max(residue_ids)
+
+    if not (min_res <= start_res <= max_res) or not (min_res <= end_res <= max_res):
+        log.error(f"❌ Error: Trimming range {trimming_range} is outside residue range of the PDB {min_res}-{max_res}")
```

Comment:
what if we have a structure with discontinuous numbering (e.g. when we do loop trimming)? The user could select a range that is not included in the PDB

I think it would be more robust to check if all the residues in the range can be found in the PDB explicitly